src/main/java/com/ait/lienzo/client/core/shape/decorator/IShapeDecorator.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape.decorator;
```

Comment:
IMO I would refactor this new decorators stuff (interface + impls) into some package inside `core.shape.wires`, rather than inside `core.shape`, because:
- I don't think, at least for now, it makes sense to have "states" in core shapes
-- Core shapes have attributes, which you can change them by the public API, and are serializable. Shape "states" are not.
-- There exist no containment, connection, etc related logic in the core, this is pure wires stuff, so decorators and shape "states" doesn't play any role, isn't it?
- In fact I guess this is only being used by wires stuff (MagnetManager,etc), right?

Well as usually just giving my opinion and brainstorming... WDYT? :)